hnn_core/network.py

```diff
@@ -764,6 +793,254 @@ def _get_src_type_and_pos(self, gid):
 
         return src_type, src_pos, src_type in self.cellname_list
 
+    # connections:
+    # this NODE is aware of its cells as targets
+    # for each syn, return list of source GIDs.
+    # for each item in the list, do a:
+    # nc = pc.gid_connect(source_gid, target_syn), weight,delay
+    # Both for synapses AND for external inputs
+    def _default_connect(self):
+        nc_dict = {
+            'A_delay': 1.,
+            'threshold': self.params['threshold'],
+        }
```

Comment:
Idk how the rest of you feel, but it might make sense to make `A_delay` and `threshold` attributes of `Network` so that they don't have to be defined for each connection individually:

1. to the best of my knowledge, these values have always been constant across a network
2. intuitively, the delay between two cells should be defined by the distance between them - our code should reflect this in as simple a way as possible

Reply:
perhaps! Would you mind opening a separate issue though? I think this PR needs to converge now ...

Reply:
I can do that, though I was just imaging that this would resolve some of the previous comments and simplify the API for a few of @ntolley's methods introduced in this PR.

Reply:
I've made both attributes, but I am keeping `self.params['threshold']` wherever it crops up to maintain functionality with HNN-GUI

Reply:
Turns out `delay` needs to be passed as a parameter for the drives API. But threshold is still fine as an attribute.